demos/http/http_demo_s3_download_async/demo_config.h

```diff
@@ -0,0 +1,125 @@
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef DEMO_CONFIG_H_
+#define DEMO_CONFIG_H_
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Logging config definition and header files inclusion are required in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for DEMO.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for DEMO.
+ */
+
+/* Include header that defines log levels. */
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "DEMO"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+#include "logging_stack.h"
+
+/************ End of logging configuration ****************/
+
+
+/**
+ * @brief HTTP server port number.
+ *
+ * In general, port 443 is for TLS HTTP connections.
+ */
+#ifndef HTTPS_PORT
+    #define HTTPS_PORT    443
+#endif
+
+/**
+ * @brief Path of the file containing the server's root CA certificate for TLS authentication.
+ *
+ * @note S3 uses the Baltimore Cybertrust root CA certificate. To download this certificate, see
+ * https://baltimore-cybertrust-root.chain-demos.digicert.com/info/index.html
+ * This path can be set with S3_ROOT_CA_CERT_PATH in cmake.
+ */
+#ifndef ROOT_CA_CERT_PATH
+    #define ROOT_CA_CERT_PATH    ""
+#endif
+
+/**
+ * @brief Presigned URL generated by python script in common/src/presigned_urls_gen.py
+ *
+ * @note This script requires AWS CLI to be configured. For instructions, see
+ * https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html
+ *
+ * Run this script and paste the output value IOT_DEMO_HTTPS_PRESIGNED_GET_URL into
+ * S3_PRESIGNED_URL below.
+ */
+#ifndef S3_PRESIGNED_URL
+    #define S3_PRESIGNED_URL    ""
+#endif
+
+/**
+ * @brief Size of file to be downloaded from S3
+ *
+ * @note This script requires AWS CLI to be configured. For instructions, see
+ * https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html
+ *
+ * Run this script and paste the output value IOT_DEMO_HTTPS_PRESIGNED_GET_URL into
+ * S3_PRESIGNED_URL below.
+ */
+#ifndef S3_FILE_SIZE
+    #define S3_FILE_SIZE    ( 0 )
+#endif
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS    ( 5000 )
+
+/**
+ * @brief The length in bytes of the user buffer.
+ */
+#define USER_BUFFER_LENGTH                ( 4096 )
+
+/**
+ * @brief The maximum length of a host address.
+ */
+#define MAX_HOST_ADRESS_LENGTH            ( 100 )
+
+/**
+ * @brief The name to use for the request queue.
+ */
+#define REQUEST_QUEUE                     "/demo_request_queue"
```

Comment:
Should this be configured? Maybe the customer doesn't care what the queues are called, it doesn't affect the platform unless the name is super long.